### PR TITLE
Run Utreexo integration tests

### PR DESCRIPTION
**Pull request by m-kus**
This PR introduces a third test type — "utreexo".
It is the same as full block tests but with Utreexo data section that includes initial accumulator state, inclusion proofs, and expected accumulator state.

The test program args are extended with Utreexo args in a backwards compatible manner, i.e. if the Utreexo section is missing, the program would run in "full" mode.

The utreexo test data generation is a bit different:
- First the script indexes from genesis up to the specified block height
- Only last block is included in the test
- This is necessary to obtain the Utreexo state **before** the update

### CODE REVIEW

File: scripts/data/regenerate_tests.sh [hunk 1]
```diff
@@ -63,4 +68,9 @@ done
 for test_case in "${full_test_cases[@]}"; do
     echo "Generating test data: full mode, chain state @ $test_case, single block"
     generate_test "full_fast" $test_case
```
Comment (maciejka) [inline scripts/data/regenerate_tests.sh:70]: `full_fast` or `full`?

File: scripts/data/utreexo.py [hunk 2]
```diff
@@ -11,27 +11,15 @@ def __init__(self, key, left=None, right=None):
         self.right = right
         self.parent = None
 
+    def __repr__(self) -> str:
+        return str(self.val)
+
 
 class Utreexo:
```
Comment (maciejka) [inline scripts/data/utreexo.py:18]: Is it a representation of Utreexo accumulator?

File: scripts/data/regenerate_tests.sh (hunk 1 above)
Comment (m-kus) [reply scripts/data/regenerate_tests.sh]: `full`, fast mode requires downloading data which does not make much sense for tests (at least for now)

File: packages/consensus/src/types/utreexo.cairo [hunk 3]
```diff
@@ -49,13 +49,31 @@ pub struct UtreexoState {
...
-            self.add(leave);
+    fn validate_and_apply(
+        ref self: UtreexoState, utxo_set: UtxoSet, proofs: Span<UtreexoProof>
+    ) -> Result<(), ByteArray> {
+        let mut lead_idx = 0;
+        let mut inner_result = Result::Ok(());
+
+        for leaf_hash in utxo_set
+            .leaves_to_delete {
+                let proof = proofs[lead_idx];
```
[the remark this answers is not in the document]
Comment (m-kus) [reply packages/consensus/src/types/utreexo.cairo]: Right, should be incremented

**Review by maciejka — APPROVED** [review]

**Pull request merged**